projects/ngx-favicon/src/lib/ngx-favicon.service.ts

```diff
@@ -1,6 +1,53 @@
-import { Injectable } from '@angular/core';
+import { Injectable, Renderer2, ElementRef, Inject } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { NGX_FAVICON_CONFIG } from './ngx-favicon.tokens';
+import { NgxFaviconConfig } from './ngx-favicon.interface';
 
+class NgxFaviconUnknownFaviconError extends Error {
+  constructor(faviconName: string) {
+    super(
+      `Favicon with name "${faviconName}" was not found into the "custom" property of the config`,
+    );
+  }
+}
+
+// https://github.com/angular/angular/issues/20351#issuecomment-344009887
+/** @dynamic */
 @Injectable()
-export class NgxFaviconService {
-  constructor() {}
+export class NgxFaviconService<CustomFavicon> {
+  constructor(
+    @Inject(DOCUMENT) private document: Document,
+    @Inject(NGX_FAVICON_CONFIG)
+    private ngxFaviconConfig: NgxFaviconConfig<CustomFavicon>,
+  ) {}
+
+  public setCustomFavicon(faviconName: CustomFavicon): void {
+    if (!faviconName) {
+      throw new NgxFaviconUnknownFaviconError(
+        (faviconName as unknown) as string,
+      );
+    }
+
+    this.updateFavicon(faviconName);
+  }
+
+  public setDefaultFavicon(): void {
+    this.updateFavicon();
+  }
+
```

Comment:
It would be handy to be able to also have an api that lets you assign an `Observable<CustomFavicon | null>` that subscribes and sets the custom or default icon. Possibly even pushes to an array and this library can `merge(...iconObservables`

Reply:
Talked about that IRL and this is not the responsibility of the lib